Add tooltip to contacts

muzakparov opened the pull request:
Related to #8790 


Contact page:

![contact](https://user-images.githubusercontent.com/26898520/87384431-bd83f080-c569-11ea-9de7-cac04dd5fa4f.gif)


Seed phrase page:

![seed](https://user-images.githubusercontent.com/26898520/87384676-574b9d80-c56a-11ea-8fb4-1a34f2d35d93.gif)

Not sure if it is necessary, but I think that copy logic could be abstracted to hooks since it seems to repeat in different places.

muzakparov committed:
add tooltip to contacts

muzakparov committed:
add copied to seed phrase

whymarrh committed:
done

Gudahtt commented on ui/app/components/ui/export-text-container/export-text-container.component.js:
```diff
@@ -5,25 +5,33 @@ import { exportAsFile } from '../../../helpers/utils/util'
-            onClick={() => copyToClipboard(text)}
+            onClick={() => {
+              this.setState({ copied: true })
+              setTimeout(() => this.setState({ copied: false }), 3000)
```
This will throw an error in the console if this component is unmounted when this timer resolves. It's a harmless error in this case, but still nice to avoid. It could be avoided by saving the timeout ID, and clearing the timeout on unmount.

Gudahtt replied:
The same applies to the similar change made to the `ViewContact` component as well

Gudahtt commented on ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js:
```diff
@@ -48,28 +66,41 @@ export default class ViewContact extends PureComponent {
+                <button
+                  className="address-book__view-contact__group__static-address--copy-icon"
+                  onClick={() => {
+                    const { checkSummedAddress } = this.props
```
Nit: It would be better to avoid shadowed variables if possible. In this case there's no need to declare another `checkSummedAddress` variable here, as there's already one in-scope.

```suggestion
```

Gudahtt commented on ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js:
```diff
@@ -48,28 +66,41 @@ export default class ViewContact extends PureComponent {
-                className="address-book__view-contact__group__static-address--copy-icon"
-                onClick={() => copyToClipboard(checkSummedAddress)}
+              <Tooltip
+                wrapperClassName="address-book__tooltip-wrapper"
```
It looks like this class doesn't exist? :thinking: If so, better to not reference it.
```suggestion
```

Gudahtt commented on ui/app/pages/settings/contact-list-tab/index.scss:
```diff
@@ -21,7 +21,7 @@
     border-bottom: 1px solid #dedede;
 
     &:hover {
-      border: 1px solid #037DD6;
```
A lot of these changes seem unrelated to the primary purpose of this PR. Is this the work of some auto-formatter or something? :thinking: 

This particular change (lower-casing hex colours) is in-line with our current stylelint config (which is broken but will be fixed shortly), but some of the other changes (e.g. adding a leading zero to decimal numbers) are in conflict with it.

Gudahtt replied:
stylelint has been re-renabled for all auto-fixable rules in #8989, which was just merged. So most of these changes should vanish now if you rebase on `develop`,

Gudahtt reviewed (commented):
Thanks @muzakparov! Looks pretty good overall - I had a few comments/questions.

>Not sure if it is necessary, but I think that copy logic could be abstracted to hooks since it seems to repeat in different places.

Definitely not necessary, but I agree that would be great. We'd just need to convert these to functional components first, so we could use hooks.

muzakparov committed:
rm junk

muzakparov committed:
add hooks

muzakparov committed:
change export default

muzakparov committed:
add hooks

muzakparov commented:
thank you @Gudahtt 
refactored to function components and added two react hooks, now it clears timeout on unmount
I am running `yarn lint:fix` before committing my changes

Gudahtt commented:
Hey @muzakparov, we only just merged the PR that included uncommenting that linting rule. After you update this branch, it should work properly if you try linting again. Sorry for the confusion!

muzakparov commented:
Cool stuff! I merged with your changes and seems to be fine now. Also, github seems to be having some synchronization issues at the moment, so I am unable to see my changes on the Pull Request.

muzakparov commented:
Linting seems a bit different on CI/CD, I had to add a comma to make the linting test pass.

Gudahtt commented:
Oh, that might be because we recently updated `eslint` in #8978, which resulted in `eslint` requiring dangling commas for parameters as well. You might need to run `yarn` locally to get the updated version, or you might be running a different version of `eslint` that is installed globally or in your editor.

muzakparov commented:
I merged with develop branch, didn't notice that I had many commits sorry

muzakparov commented:
@Gudahtt could you please help me with the error:)

Gudahtt commented:
Sure, no problem! I'll take a look

Gudahtt commented:
I'm not sure exactly what went wrong, but I backtracked and merged `develop` into this branch again, and now it seems to be OK!

Sorry for the delay in reviewing this - I'm hoping to get to it today or tomorrow.

Gudahtt commented on ui/app/hooks/useTimeout.js:
```diff
@@ -0,0 +1,40 @@
+ *
+ * @param {Function}  cb   - callback function inside setTimeout
+ * @param {number}  delay   - delay in ms
+ * @param {boolean}  immediate   - delay in ms
```
This should be marked as optional, and the description should be updated. Something like this perhaps:
```suggestion
 * @param {boolean}  [immediate] - determines whether the timeout is invoked immediately
```

Gudahtt commented on ui/app/hooks/useTimeout.js:
```diff
@@ -0,0 +1,40 @@
+    }
+  }, [delay, triggered])
+
+  const trigger = useCallback(() => {
```
Nit: I was a bit confused by this name at first; could this be changed to something like `startTimeout`?

Gudahtt commented on ui/app/hooks/useTimeout.js:
```diff
@@ -0,0 +1,40 @@
+  }, [delay, triggered])
+
+  const trigger = useCallback(() => {
+    setTriggered(true)
```
Note that this wouldn't reset the timer if the user copied the address multiple times. This could lead to strange behavior, e.g. the text rapidly flashing between states, rather than showing for 3 seconds.

Perhaps you could store the timeout id in state instead of this `triggered` boolean? Then the old timeout could be cleared here (if it's set), allowing the timer to be reset.

muzakparov replied:
thank you for the suggestions, now using `timeoutId` in a state, so that it is cleared/reset whenever `startTimeout` is called

Gudahtt reviewed (commented):
These changes look fantastic! I had a couple of follow up suggestions/questions, let me know what you think.

muzakparov committed:
minor fixes

muzakparov committed:
merge again

muzakparov committed:
rm spaces

muzakparov committed:
lint

Gudahtt reviewed (approved):
LGTM, great work!
